packages/slice-machine/src/models/slice/context.tsx

```diff
@@ -126,14 +92,9 @@ export const SliceHandler = ({ children }: { children: any }) => {
 
   // variation not in the URL but a default variation was found
   if (!variationParam) {
-    void router.replace(`/${lib.name}/${slice[0].model.name}/${variation.id}`);
+    void router.replace(`/${lib.name}/${slice.model.name}/${variation.id}`);
   }
 
-  return (
-    <SliceProvider value={slice} variation={variation}>
-      {children}
-    </SliceProvider>
-  );
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-call
+  return <>{typeof children === "function" ? children(slice) : children}</>;
```

Comment:
you can't return `children` as it is since it's a `React.ReactNode`, while a react component expects a `ReactElement`. Placing it within a fragment fixes it since it will make it return an element (https://github.com/DefinitelyTyped/DefinitelyTyped/issues/33006)

same thing is done elsewhere in the app I see
<img width="482" alt="image" src="https://user-images.githubusercontent.com/69144671/177304230-b9313b34-c052-4722-ac3f-708769403cba.png">

Reply:
think it's because a ReactNode could be a function that tale slice as param? I am not too sure tbh, it was already there and didn't wanna edit something I don't fully understand

Reply:
that's understandable :)

Reply:
funny, never this issue before.

Reply:
Do you know why it suddenly triggered an issue ?

Reply:
don't think it triggered any issues, think it's just something a bit odd that both me and Marc don't seem to be too sure so it raised a question when I moved it and it showed up in the git diff